tasks/managed/apply-mapping/apply-mapping.yaml

```
@@ -395,7 +396,28 @@ spec:
                 oci_version_raw="$(jq -r \
                   '.Labels."org.opencontainers.image.version" // ""'\
                   <<< "$image_metadata")"
-                labels="$(jq -c '.Labels' <<< "${image_metadata}")"
+                env_variables="$(jq -c '.Env // []' <<< "${image_metadata}")"
+                labels="$(jq -c '.Labels // {}' <<< "${image_metadata}")"
+            fi
+
+            # Add image env_variables metadata to component
+            if [ "$(jq 'length' <<< "$env_variables")" -ne 0 ] ; then
+              env_file=$(mktemp)
+              echo "$env_variables" > "$env_file"
+              jq --argjson i "$i" --slurpfile env "$env_file" \
+                '.components[$i].metadata = (.components[$i].metadata // {}) * {env_variables: $env[0]}' \
+                "${SNAPSHOT_SPEC_FILE}" > /tmp/temp && mv /tmp/temp "${SNAPSHOT_SPEC_FILE}"
+            fi
+
+            # Add image labels metadata to component
+            if [ "$(jq 'length' <<< "$labels")" -ne 0 ] ; then
+              # Convert labels from {key: value} to [{name: key, value: value}]
+              labels_array=$(jq -c 'if . then to_entries | map({name: .key, value: .value}) else [] end' <<< "$labels")
+              labels_file=$(mktemp)
+              echo "$labels_array" > "$labels_file"
```

```suggestion
              labels_file=$(mktemp)
              jq -c 'if . then to_entries | map({name: .key, value: .value}) else [] end' <<< "$labels" > "$labels_file"
              
```
Seems more straightforward maybe? Plus one less line :)

---

Fixed TY :)